server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/project/ProjectService.java

```diff
@@ -109,8 +114,50 @@ public class ProjectService {
     @Autowired
     private DatabaseRepository databaseRepository;
 
+    @Autowired
+    private ResourceRoleRepository resourceRoleRepository;
+
+    @Autowired
+    private UserResourceRoleRepository userResourceRoleRepository;
 
-    private ProjectMapper projectMapper = ProjectMapper.INSTANCE;
+    @Autowired
+    private ConnectionConfigRepository connectionConfigRepository;
+
+    @Autowired
+    private ConnectionService connectionService;
+
+    private final ProjectMapper projectMapper = ProjectMapper.INSTANCE;
+
+    @SkipAuthorize("odc internal usage")
+    @Transactional(rollbackFor = Exception.class)
+    public void createProjectIfNotExists(@NotNull User user) {
+        String projectName = user.getAccountName();
+        if (repository.findByNameAndOrganizationId(projectName, user.getOrganizationId()).isPresent()) {
+            return;
+        }
+        ProjectEntity projectEntity = new ProjectEntity();
+        projectEntity.setBuiltin(true);
+        projectEntity.setArchived(false);
+        projectEntity.setName(projectName);
+        projectEntity.setCreatorId(user.getCreatorId());
+        projectEntity.setLastModifierId(user.getCreatorId());
+        projectEntity.setOrganizationId(user.getOrganizationId());
+        projectEntity.setDescription("Built-in project for bastion user " + user.getAccountName());
+        ProjectEntity saved = repository.saveAndFlush(projectEntity);
+        // Grant DEVELOPER role to bastion user, and all other roles to user creator(admin)
+        List<UserResourceRoleEntity> userResourceRoleEntities = ResourceRoleName.all().stream().map(name -> {
+            ResourceRoleEntity resourceRoleEntity =
+                    resourceRoleRepository.findByResourceTypeAndRoleName(ResourceType.ODC_PROJECT, name)
```

Comment:
Prefer to load the resource roles from the metadb at once rather than load one by one in iteration.